app/api/api_v1/routers/search.py

```diff
@@ -53,7 +53,7 @@
 def _search_request(
     db: Session, search_body: SearchRequestBody, use_vespa: bool = False
 ) -> SearchResponse:
-    if search_body.keyword_filters is not None:
+    if search_body.keyword_filters is not None and not use_vespa:
```

Comment:
The double negative might read easier with: 
```suggestion
    if search_body.keyword_filters is not None and use_vespa == False:
```
What do you think?